studio/components/interfaces/Integrations/ProjectLinker.tsx

```diff
@@ -79,9 +79,18 @@ const ProjectLinker = ({
   const [supabaseProjectRef, setSupabaseProjectRef] = useState<string | undefined>(
     defaultSupabaseProjectRef
   )
+  useEffect(() => {
+    if (defaultSupabaseProjectRef !== undefined && supabaseProjectRef == undefined)
+      setSupabaseProjectRef(defaultSupabaseProjectRef)
+  }, [defaultSupabaseProjectRef, supabaseProjectRef])
+
```

Comment:
Couldn't all this logic just go directly in the useState initialState?

Reply:
That would be nice, but `defaultSupabaseProjectRef` is `undefined `initially thanks to it being a next.js query param

Reply:
gotcha